### ISSUE TITLE
Add missing IPhysicsObject methods

### ISSUE THREAD

**Issue by jcwik223** [Facepunch/garrysmod-requests#998]
I'm currently working on some projects that heavily involve physics objects (portal-esque gun, Lua grav and physguns) and I'm a bit stuck without a few of these methods. I put the more important ones at the top.

// AddAngleVelocity seems to treat the addition as an applicative force and is not set immediately which is why I'm guessing the Instantaneous version exists
* [ ] SetAngleVelocity (2nd arg of ::SetVelocity)
* [ ] SetAngleVelocityInstantaneous (2nd arg of ::SetVelocityInstantaneous)
* [x] ApplyTorqueCenter (for rotational kinematics on an arbitrary point instead of general rotation)
* [x] GetShadowPosition (used by gravgun in simulating motion on a held object)
* [x] GetVelocityAtPoint (can be used to find a close frictional force estimate)
* [ ] Get/SetCallbackFlags (need CALLBACK_GLOBAL_TOUCH_STATIC for an NPC)
* [ ] GetSphereRadius (can be used to see if spherical physobjs are inside one another)
* [ ] SetShadow (like UpdateShadow but will maintain the current position/angle data while only updating speed and physics behavior)

-------------------------------------------------------------

* [x] Get/SetContents (setting what a physobj is made of can allow entities to have complex collision behavior depending on each physobj)
* [ ] Get/SetGameData (usually used to link constraint physobjs to the entity being constrained, which has some internal optimization properties in regards to sleeping and collision)
* [ ] Get/SetGameFlags (used to disable physobj self-collisions, pickup ability and a number of other things; see FVPHYSICS flags)
* [ ] IsHinged/BecomeHinged/RemoveHinged (can be used to isolate object to one axis of movement)
* [ ] CalculateLinearDrag/CalculateAngularDrag
* [ ] IsStatic/IsFluid/IsTrigger (telling the physics object type for exact recreation)
* [ ] IsAttachedToConstraint

**Comment by thegrb93** [comment]
Yes. Torque is so needed. The current method is probably 100x slower.

**Comment by Kefta** [comment]
Related to https://github.com/Facepunch/garrysmod-requests/issues/944

**Comment by robotboy655** [comment]
Added PhysObj.ApplyTorqueCenter( Vector/AngularImpulse )
Added PhysObj.GetShadowPos() = Vector
Added PhysObj.GetShadowAngles() = Angle
Added PhysObj.GetVelocityAtPoint( Vector ) = Vector
Added PhysObj.SetContents( uint )
Added PhysObj.GetContents() = uint
